interface/wx/msw/registry.h

```diff
@@ -34,28 +35,29 @@
     @b Example:
 
     @code
-    // This assume that the key already exists, use HasSubKey() to check
+    // This assumes that the key already exists, use HasSubKey() to check
     // for the key existence if necessary.
-    wxRegKey key(wxRegKey::HKLM, "Software\\MyKey");
+    wxRegKey key(wxRegKey::HKCU, "Software\\MyKey");
 
     // Create a new value "MyValue" and set it to 12.
     key.SetValue("MyValue", 12);
 
     // Read the value back.
     long value;
     key.QueryValue("MyValue", &value);
-    wxMessageBox(wxString::Format("%d", value), "Registry Value", wxOK);
+    wxLogMessage("Registry value: %ld", value);
 
-    // Get the number of subkeys and enumerate them.
-    size_t subkeys;
-    key.GetKeyInfo(&subkeys, NULL, NULL, NULL);
+    // Enumerate the subkeys.
+    wxString keyName;
+    long index = 0;
 
-    wxString key_name;
-    key.GetFirstKey(key_name, 1);
-    for(int i = 0; i < subkeys; i++)
+    if ( key.GetFirstKey(keyName, index) )
     {
-        wxMessageBox(key_name, "Subkey Name", wxOK);
-        key.GetNextKey(key_name, 1);
+        do
+        {
+            wxLogMessage("Subkey name: %s", keyName);
+        }
+        while ( key.GetNextKey(keyName, index) );
```

Comment:
Considering how rare `do...while` loops are, I'm not sure it's worth using them in an example. Would there be any problem with doing
```cpp
for ( bool cont = key.GetFirstKey(keyName, index); cont; cont = key.GetNextKey(keyName, index) )
{
    ...
}
```
instead?

Reply:
The original reason why I decided to change the enumeration code was that it used literal `1` where a reference to a `long` (index) was expected, making the code both not only incorrect but also uncompilable.

When I was at it, I changed the algorithm to use what I thought was the usual implementation of the findfirst/findnext API idiom - with a condition check at the end of the loop. But if you find the implementation using `for` to be better, I am going to change the code accordingly.

BTW, I have noticed that headers of some files still have your e-mail address at dptmaths.ens-cachan.fr but I guess that’s preserved for historical reasons and unlike the wxwindows.org one does not need to be changed to point to wxwidgets.org.

Reply:
Oh, and forgot to answer this one: yes, that email is very old and doesn't work since ~3 decades, but I don't think it's worth changing it, it should be easy enough to find the working one anyhow.